http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -669,6 +669,56 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
             .raiseError[Unit](AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "permission denied"))
     } yield GetAppResponse.fromDbResult(app, Config.proxyConfig.proxyUrlBase)
 
+  override def updateAppConfig(userInfo: UserInfo,
+                               cloudContext: CloudContext.Gcp,
+                               appName: AppName,
+                               req: UpdateAppConfigRequest
+  )(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      // throw 403 if no project-level permission
+      hasProjectPermission <- authProvider.isUserProjectReader(
+        cloudContext,
+        userInfo
+      )
+      _ <- F.raiseWhen(!hasProjectPermission)(ForbiddenError(userInfo.userEmail, Some(ctx.traceId)))
+
+      // convert from Option[Boolean] to Boolean
+      autodeleteEnabled = req.autodeleteEnabled.getOrElse(false)
+      // if autodeleteEnabled, then a positive autodeleteThreshold is required
+      isValidConfiguration = !autodeleteEnabled || (req.autodeleteThreshold.getOrElse(0) > 0)
+      _ <- F.raiseWhen(!isValidConfiguration)(
+        BadRequestException("autodeleteThreshold must be positive if autodeleteEnabled", Some(ctx.traceId))
+      )
+
+      appOpt <- KubernetesServiceDbQueries
+        .getActiveFullAppByName(cloudContext, appName)
+        .transaction
+      appResult <- F.fromOption(
+        appOpt,
+        AppNotFoundException(cloudContext, appName, ctx.traceId, "No active app found in DB")
+      )
+
+      tags = Map("appType" -> appResult.app.appType.toString)
+      _ <- metrics.incrementCounter("updateAppConfig", 1, tags)
```

Comment:
I noticed that the counter is sometimes incremented at the Service level and sometimes at the Routes level.  Unclear which to choose.